### PR TITLE
feat!: transfer cf action

### PR DESCRIPTION
**Pull request by bharathkkb**
- Transfer action from https://github.com/google-github-actions/setup-gcloud/tree/master/deploy-cloud-functions

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Cloud Functions Deploy
+description: |-
+  Cloud Functions is a scalable Functions-as-a-Service (FaaS) to run your 
+  code with zero server management. Use this action to deploy your function or
+  update an existing Cloud Function.
+author: GoogleCloudPlatform
```
Comment (averikitsch) [inline action.yaml:20]: nit: `Google LLC` may make more sense here

File: package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "url": "https://github.com/google-github-actions/deploy-cloud-functions"
+  },
+  "keywords": [
+    "actions",
+    "gcf",
+    "functions",
+    "google cloud",
+    "cloud function"
+  ],
+  "author": "GoogleCloudPlatform",
```
Comment (averikitsch) [inline package.json:31]: same here

**Review by averikitsch — APPROVED** [review]

File: .github/workflows/deploy-cf-it.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  gcloud:
+    name: with setup-gcloud
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master # Set up ADC for credentials
```
Comment (averikitsch) [inline .github/workflows/deploy-cf-it.yml:11]: update

File: action.yaml (hunk 1 above)
Comment (bharathkkb) [reply action.yaml]: done

File: package.json (hunk 2 above)
Comment (bharathkkb) [reply package.json]: done

File: .github/workflows/deploy-cf-it.yml (hunk 3 above)
Comment (bharathkkb) [reply .github/workflows/deploy-cf-it.yml]: done

**Pull request merged**